gssapi/tests/test_raw.py

```diff
@@ -644,6 +644,66 @@ def test_add_cred_with_password(self):
 
         new_creds.should_be_a(gb.Creds)
 
+    @ktu.gssapi_extension_test('rfc5587', 'RFC 5587')
+    def test_rfc5587(self):
+        mechs = gb.indicate_mechs_by_attrs(None, None, None)
+        mechs.shouldnt_be_empty()
+        mechs.should_be_a(set)
+
+        last_mech = None
```

Comment:
Can you just use `mechs[-1]` instead of repeatedly updating a variable for this?